File: 0066-plus-one/0066-plus-one.cpp
class Solution {
public:

    vector<int> plusOne(vector<int>& digits) {
        
        int n=digits.size();
        int unit=digits[n-1];
        
        if(unit<9){
            digits[n-1]++;
            return digits;
        }else{
            int i=n-1;
            while(i>=0 && digits[i] ==9){
                i--;
            }//   i      
            // 20539999
            if(i==-1){
                vector<int> ans(n+1,0);
                ans[0]=1;
                return ans;
            }else{
                digits[i]++;
                for(int j=i+1;j<n;j++){
                    digits[j]=0;
                }
            }
        }
        return digits;
    }
};

File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        
        int n=digits.length;

        boolean extraDigit=false;

        if(digits[n-1]==9){
            int i=n-1;
            while(i>=0 && digits[i]==9){
                digits[i]=0;
                i--;
            }
            if(i<0) extraDigit=true;
            else digits[i]++;
        }else{
            digits[n-1]++;
        }


        if(extraDigit==true){
            int[] ans= new int[n+1];
            ans[0]=1;

            for(int i=1;i<=n;i++){
                ans[i]=digits[i-1];
            }
            return ans;
        }

        return digits;
    }
}